moveit_ros/planning/trajectory_execution_manager/src/trajectory_execution_manager.cpp

```diff
@@ -1430,6 +1437,61 @@ bool TrajectoryExecutionManager::executePart(std::size_t part_index)
   }
 }
 
+bool TrajectoryExecutionManager::waitForRobotToStop(const TrajectoryExecutionContext &context, double wait_time)
+{
+  if (allowed_start_tolerance_ == 0)  // skip validation on this magic number
+    return true;
+
+  ros::WallTime start = ros::WallTime::now();
+  double time_remaining = wait_time;
+
+  robot_state::RobotStatePtr prev_state, cur_state;
+  prev_state = csm_->getCurrentState();
+  prev_state->enforceBounds();
+
+  // assume robot stopped when 3 consecutive checks yield the same robot state
+  unsigned int no_motion_count = 0;  // count iterations with no motion
```

Comment:
std::size_t is required for array access. Here we count up to 3 only. No need for a 64bit integer ;-)